src/magic.cpp

```diff
@@ -1423,23 +1425,57 @@ int known_magic::get_spellname_max_width()
     return width;
 }
 
+static bool is_valid_invlet( const int invlet )
+{
+    if( invlet >= 'a' && invlet <= 'z' ) {
+        return true;
+    }
+    if( invlet >= 'A' && invlet <= 'Z' ) {
+        return true;
+    }
+    if( invlet >= '!' && invlet <= '-' ) {
+        return true;
+    }
+    return false;
+}
+
 class spellcasting_callback : public uilist_callback
 {
     private:
         std::vector<spell *> known_spells;
         void draw_spell_info( const spell &sp, const uilist *menu );
     public:
+        // invlets reserved for special functions
+        const std::set<int> reserved_invlets{ 'I', '=' };
         bool casting_ignore;
 
         spellcasting_callback( std::vector<spell *> &spells,
                                bool casting_ignore ) : known_spells( spells ),
             casting_ignore( casting_ignore ) {}
-        bool key( const input_context &, const input_event &event, int /*entnum*/,
+        bool key( const input_context &, const input_event &event, int entnum,
                   uilist * /*menu*/ ) override {
             if( event.get_first_input() == 'I' ) {
                 casting_ignore = !casting_ignore;
                 return true;
             }
+            if( event.get_first_input() == '=' ) {
+                int invlet = 0;
+                invlet = popup_getkey( _( "Choose a new hotkey for this spell." ) );
+                if( is_valid_invlet( invlet ) ) {
+                    const bool invlet_set = g->u.magic.set_invlet( known_spells[entnum]->id(), invlet,
+                                            reserved_invlets );
+                    if( !invlet_set ) {
+                        popup( _( "Hotkey already used." ) );
+                    } else {
+                        popup( _( "%s set. Close and reopen spell menu to refresh list with changes." ),
```

Comment:
```suggestion
                        popup( _( "%c set.  Close and reopen spell menu to refresh list with changes." ),
```

Reply:
i tried that, but it didn't do it correctly. are you sure this functionality currently exists with _()?